### PR TITLE
Follow up to #57568

### PR DESCRIPTION
**Pull request by kssenii**
### Changelog category (leave one):
- Not for changelog (changelog entry is not required)

Follow-up to https://github.com/ClickHouse/ClickHouse/pull/57568.

### CODE REVIEW

**Comment by robot-ch-test-poll4** [comment]
*This is an automated comment for commit a4a264c78561a37d034231e4ef00356d2925b586 with description of existing statuses. It's updated for the latest CI running*

[⏳ Click here](https://s3.amazonaws.com/clickhouse-test-reports/57863/a4a264c78561a37d034231e4ef00356d2925b586/ci_running.html) to open a full report in a separate page

<details><summary>Successful checks</summary>
<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>Mergeable Check</td><td>Checks if all other necessary checks are successful</td><td><a href="https://github.com/ClickHouse/ClickHouse/actions/runs/7209277830">✅ success</a></td></tr>
<tbody>
</table>
</details>
<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>CI running</td><td>A meta-check that indicates the running CI. Normally, it's in <b>success</b> or <b>pending</b> state. The failed status indicates some problems with the PR</td><td><a href="https://s3.amazonaws.com/clickhouse-test-reports/57863/a4a264c78561a37d034231e4ef00356d2925b586/ci_running.html">⏳ pending</a></td></tr>
<tbody>
</table>

File: src/Storages/PostgreSQL/MaterializedPostgreSQLConsumer.cpp [hunk 1]
```diff
@@ -138,15 +138,35 @@ MaterializedPostgreSQLConsumer::StorageData::Buffer::Buffer(
         columns_ast.children.emplace_back(std::make_shared<ASTIdentifier>(name));
 }
 
-MaterializedPostgreSQLConsumer::StorageData::Buffer & MaterializedPostgreSQLConsumer::StorageData::getBuffer()
+MaterializedPostgreSQLConsumer::StorageData::Buffer & MaterializedPostgreSQLConsumer::StorageData::getLastBuffer()
 {
-    if (!buffer)
+    if (!buffers.empty())
```
Comment (tavplubix) [inline src/Storages/PostgreSQL/MaterializedPostgreSQLConsumer.cpp:143]: Did you mean `if (buffers.empty())`?

Comment (kssenii) [reply src/Storages/PostgreSQL/MaterializedPostgreSQLConsumer.cpp]: yep, thx

**Review by tavplubix — APPROVED** [review]

**Comment by kssenii** [comment]
> Integration tests 

`test_storage_hdfs/test.py::test_write_table` - flaky in master
`test_storage_azure_blob_storage/test_cluster.py::test_unset_skip_unavailable_shards` https://github.com/ClickHouse/ClickHouse/issues/57889

**Pull request merged**